lib/ansible/modules/monitoring/pagerduty_alert.py

```diff
@@ -153,11 +153,17 @@ def check(module, name, state, service_id, integration_key, api_key, incident_ke
     if info['status'] != 200:
         module.fail_json(msg="failed to check current incident status."
                              "Reason: %s" % info['msg'])
-    json_out = json.loads(response.read())["incidents"][0]
 
-    if state != json_out["status"]:
-        return json_out, True
-    return json_out, False
+    incidents = json.loads(response.read())["incidents"]
+
+    if len(incidents) == 0 and state in ('acknowledged', 'resolved'):
```

Comment:
How about this - 

```suggestion
    msg = "No corresponding incident"
    if len(incidents) == 0:
        if state in ('acknowledged', 'resolved'):
            return msg, False
        return msg, True
    elif state != incidents[0]["status"]:
        return incidents[0], True

    return incidents[0], False
```

Reply:
Hi, I added your suggestion, and also rebased on devel.